### ISSUE TITLE
Some definitions - unclear or missing

### ISSUE THREAD

**Issue by H2Swine** [ietf-wg-cellar/flac-specification#213]
Reading the draft with fresh eyes, I see a couple of potential issues with the definitions or lack thereof. Given my habit of rather asking a question too many than one too few, this could be long - I am splitting off a separate issue #214 , leaving this to focus on (other) definitions.

Let me start with section **3. Definitions.**

- The first term to be defined, is "Lossless compression", naturally.  But maybe surprisingly, the next term, "Lossy compression", is not _used_ in the document.  Does it then even belong in the Definitions section?  Text-searching for "lossy" leads to 9.2.2 with a mention of LossyWAV (that is a _name_ rather than a description, although it is a descriptive name) - and I wonder if it is even necessary to name other codecs or processors. That goes also for the "otherwise lossless" MLP. There are two occurrences of "non-lossless"; one in the same sentence, and at the end of subsection 4.2.  I will soon (edit: done!) post a new issue proposing changes to 9.2.2, but meanwhile: should "Lossy compression" just be relegated to a final sentence at the end of "Lossless compression"?
- Editing in this as the issue is only a quarter of an hour old: the "Subframe" definition first writes "Subframe: An encoded subblock" and then only after a sentence in between, it suggests the fact that it isn't really. I am not sure how to phrase it better. Attempt: "Subframe: An encoded subblock, or an encoded channel corresponding to average between two or difference between two as specified in <reference>, so that the subframes within a frame all together encode the entire block. All subframes within a frame encode the same number of samples." (I am not sure if "code for" was intentional?!)
- "audio". Naturally, that phrase is used all over - and I am not even sure if it should be defined, when it is used so loosely; but FLAC can also encode raw data. And in 8.2 it speaks about "contains audio".  
_Possibly_ one should start by saying that "Digital audio" is a sequence of time-indexed data points, which is the payload data (to be) stored in a FLAC file.  For most applications, these data represent actual sound, and so the phrase "audio" will be used even if the data is not intended for audio playback.
- "CD-DA". See below. I will discuss audio before Compact Disc Digital Audio.

**The use of "audio":**
As already mentioned, there is "audio" used synonymous with encoded data payload, whether they are sound signals or not (like, "MD5 signature of the unencoded audio data") and thre is audio in the sense of "playable audio" or "audio intended for playback" as in 8.2. 

>    The sample rate MUST NOT be 0 when the FLAC file contains audio.  A
>    sample rate of 0 MAY be used when non-audio is represented.  This is
>    useful if data is encoded that is not along a time axis, or when the
>    sample rate of the data lies outside the range that FLAC can
>    represent in the streaminfo metadata block.

Of course, if the data is actual audio but with a sample rate FLAC cannot store, nitpickers will run into a clash.  But "contains audio" still looks weird, when "audio" is throughout the document used for "data payload".  Here is one wording that may resolve it:

>    The sample rate MUST NOT be 0 when the FLAC file contains audio 
>    intended for playback.  A sample rate of 0 MAY be used when other 
>    content is represented.  This could be useful to encode data which is  
>    intended to be skipped by an audio player application, for reasons 
>    that may include not being audio, or not being sampled at rates that 
>    FLAC can represent in the streaminfo metadata block, possibly not 
>    even sampled along a time axis.

9.1.7 should/could be amended accordingly. 

Here I have presumed the following intention: _if you intend to play the audio out of the decoder, use the actual sampling rate_.  If not - possibly because the sampling rate cannot be accommodated - then you can use 0.  Sample rate of 0 serves as a flag for "do not try to actually play this". 
It also raises the question: _should a decoder be recommended to skip sample rate = 0 if it passes the data to playback?_  Literally trying to play back something with infinitely long time between samples, would make it hang. 

**Then there is CD-DA.**  As there is a "MUST" requirement on cuesheets for "CD-DA", then CD-DA had better be defined.  The question is, is an audio stream "CD-DA" whenever it is stereo, 44.1 kHz, 16 bits and sample count divisible by 588?  Even if it has never been on a CD, never been mastered for CD, never was intended to be put on CD - then by coincidence, (sample count) / 588 might end up integer.
And, is a cuesheet block never intended to be a "CD cue sheet" all for sudden invalid then? 
Doesn't sound sensible.  The sensible thing is instead that anything that reads such a "CDDA-invalid" cuesheet should (or "MUST"?!) take it to mean "this is not meant for Compact Disc Digital Audio". Stereo 44.1/16 check, total sample count divisible by 588, check - cuesheet that doesn't comply, then OK it isn't meant to be burnt to audio CD.

That also _in part_ resolves the following question, which I think should be asked: is it even up to a FLAC specification to impose those restrictions on a cuesheet (i.e. on indicated track and index boundaries)? Don't they rather follow from the appropriate-coloured book, which the FLAC format itself cares nothing about?  If you want to _interoperate_ with the CD-DA colored book format, you must have such a stream stored, and then the cuesheet must specify such track boundaries and index points - but should FLAC (the specification) care?  Should it be relegated to an appendix - say the interoperability section?

**Comment by ktmf01** [comment]
> Reading the draft with fresh eyes, I see a couple of potential issues with the definitions or lack thereof.

As long as we keep reading, we keep finding problems. At some point we'll have to accept them. I'm not sure we're already too far in the process for these kinds of comments.

> But maybe surprisingly, the next term, "Lossy compression", is not _used_ in the document.

I don't see the problem of having a definition not being used in the document. I included it mostly to contrast with lossless compression, as lossy compression is something specific to audiovisual inputs, which aren't day-to-day business in the IETF.
 
I'll get back on the other issues later.

> **The use of "audio":** As already mentioned, there is "audio" used synonymous with encoded data payload, whether they are sound signals or not (like, "MD5 signature of the unencoded audio data") and thre is audio in the sense of "playable audio" or "audio intended for playback" as in 8.2.

Maybe we should add a line to format principles or notations and conventions stating that this document uses wording referring to digital audio for simplicity, but that a FLAC file is also suited storing other, non-audio signals, like measurement data.

> **Then there is CD-DA.** As there is a "MUST" requirement on cuesheets for "CD-DA", then CD-DA had better be defined.

CD-DA cuesheets are cuesheets where the main cuesheet block says that this cuesheet belongs to a CD. That specific bit is described as

> 1 if the cuesheet corresponds to a Compact Disc, else 0

How an application writing a cuesheet determines whether a cuesheet belongs to a compact disc seems out-of-scope for this document to me.
 
> is it even up to a FLAC specification to impose those restrictions on a cuesheet

They have been there for years, I am not sure it is up to me to remove them. Adding meaning to reserved bits is one thing, but removing meaning to certain bits is something else.

**Comment by H2Swine** [comment]
> As long as we keep reading, we keep finding problems. At some point we'll have to accept them. 

Totally agree. The cost/benefit considerations here are up to ... someone's discretion, not mine.

I'll just quote myself: 

>> Given my habit of rather asking a question too many than one too few


And, concerning the "corresponds to a Compact Disc" ... I should have text searched for "Compact Disc" rather than just "CD". Mea culpa.
